accounts/abi/bind/bind.go

```diff
@@ -0,0 +1,162 @@
+	var (
+		calls     = make(map[string]*tmplMethod)
+		transacts = make(map[string]*tmplMethod)
+	)
+	for _, original := range evmABI.Methods {
+		// Normalize the method for capital cases and non-anonymous inputs/outputs
+		normalized := original
+		normalized.Name = capitalize(original.Name)
+
+		normalized.Inputs = make([]abi.Argument, len(original.Inputs))
+		copy(normalized.Inputs, original.Inputs)
+		for i, input := range normalized.Inputs {
+			if input.Name == "" {
+				normalized.Inputs[i].Name = fmt.Sprintf("arg%d", i)
+			}
+		}
+		normalized.Outputs = make([]abi.Argument, len(original.Outputs))
+		copy(normalized.Outputs, original.Outputs)
+		for i, output := range normalized.Outputs {
+			if output.Name != "" {
+				normalized.Outputs[i].Name = capitalize(output.Name)
+			}
+		}
+		// Append the methos to the call or transact lists
+		if original.Const {
+			calls[original.Name] = &tmplMethod{Original: original, Normalized: normalized, Structured: structured(original)}
+		} else {
+			transacts[original.Name] = &tmplMethod{Original: original, Normalized: normalized, Structured: structured(original)}
+		}
+	}
+	// Generate the contract template data content and render it
+	data := &tmplData{
+		Package:     pkg,
+		Type:        capitalize(kind),
+		InputABI:    strippedABI,
+		InputBin:    strings.TrimSpace(bytecode),
+		Constructor: evmABI.Constructor,
+		Calls:       calls,
+		Transacts:   transacts,
+	}
+	buffer := new(bytes.Buffer)
+
+	funcs := map[string]interface{}{
+		"bindtype": bindType,
+	}
+	tmpl := template.Must(template.New("").Funcs(funcs).Parse(tmplSource))
+	if err = tmpl.Execute(buffer, data); err != nil {
+		return "", err
+	}
+	// Pass the code through goimports to clean it up and double check
+	code, err := imports.Process("", buffer.Bytes(), nil)
+	if err != nil {
+		return "", fmt.Errorf("%v\n%s", err, buffer)
+	}
+	return string(code), nil
+}
+
+// bindType converts a Solidity type to a Go one. Since there is no clear mapping
+// from all Solidity types to Go ones (e.g. uint17), those that cannot be exactly
+// mapped will use an upscaled type (e.g. *big.Int).
+func bindType(kind abi.Type) string {
+	stringKind := kind.String()
+
+	switch {
+	case stringKind == "address":
+		return "common.Address"
+
+	case stringKind == "hash":
+		return "common.Hash"
+
+	case strings.HasPrefix(stringKind, "bytes"):
+		if stringKind == "bytes" {
+			return "[]byte"
+		}
+		return fmt.Sprintf("[%s]byte", stringKind[5:])
+
+	case strings.HasPrefix(stringKind, "int"):
+		switch stringKind[:3] {
+		case "8", "16", "32", "64":
+			return stringKind
+		}
+		return "*big.Int"
+
+	case strings.HasPrefix(stringKind, "uint"):
+		switch stringKind[:4] {
+		case "8", "16", "32", "64":
+			return stringKind
+		}
+		return "*big.Int"
+
+	default:
+		return stringKind
+	}
+}
+
+// capitalize makes the first character of a string upper case.
+func capitalize(input string) string {
+	return strings.ToUpper(input[:1]) + input[1:]
```

Comment:
I know, but it should never be called for the empty string :D That usually means we forgot to handle a possible missing param somewhere, which will cause bigger issues later on anyway.